java/server/test/org/openqa/selenium/grid/gridui/ConsolePageTest.java

```diff
@@ -0,0 +1,131 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.grid.gridui;
+
+import static org.junit.Assert.assertNotNull;
+import static org.openqa.selenium.json.Json.MAP_TYPE;
+import static org.openqa.selenium.remote.http.HttpMethod.GET;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.ImmutableCapabilities;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.grid.commands.Standalone;
+import org.openqa.selenium.grid.config.Config;
+import org.openqa.selenium.grid.config.MemoizedConfig;
+import org.openqa.selenium.grid.config.TomlConfig;
+import org.openqa.selenium.grid.server.Server;
+import org.openqa.selenium.grid.web.Values;
+import org.openqa.selenium.net.PortProber;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.remote.http.HttpClient;
+import org.openqa.selenium.remote.http.HttpRequest;
+import org.openqa.selenium.remote.http.HttpResponse;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import static org.junit.Assert.assertEquals;
+
+import java.io.StringReader;
+import java.time.Duration;
+import java.util.Map;
+
+public class ConsolePageTest {
+
+  private static final int port = PortProber.findFreePort();
+
+  private Server<?> server;
+
+  @Before
+  public void setFields() {
+    this.server = createStandalone();
+  }
+
+  @After
+  public void stopServers() {
+    this.server.stop();
+  }
+
+  @Test
+  public void testConsolePage() {
+    Capabilities caps = new ImmutableCapabilities("browserName", "chrome");
+    WebDriver driver = new RemoteWebDriver(server.getUrl(), caps);
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2));
+
+    driver.get("localhost:" + port + "/ui/index.html#/console");
+
+    WebElement element = wait.until(ExpectedConditions.visibilityOf(driver.findElement(By.id("ring-system"))));
+
+    assertNotNull(element);
+    assertEquals("100% free", element.getText());
+  }
+
+  @Test
+  public void testNodePage() {
+    Capabilities caps = new ImmutableCapabilities("browserName", "chrome");
+    WebDriver driver = new RemoteWebDriver(server.getUrl(), caps);
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2));
+
+    driver.get("localhost:" + port + "/ui/index.html#/console");
+    WebElement element = wait.until(ExpectedConditions.visibilityOf(driver.findElement(By.xpath("//a[contains(@href,'node')]"))));
+
+    assertNotNull(element);
+
+    element.click();
+
+    WebElement nodePage = wait.until(ExpectedConditions.visibilityOf(driver.findElement(By.xpath("//*[text()='root']"))));
+
+    assertNotNull(nodePage);
```

Comment:
Same here. If we are going to assert lets make it for something that is specific to that page.

Reply:
I removed the assert, and added a unique id to the node info dropdown.